### DISCUSSION TITLE
I have a revealjs presentation. I want a web page displaying all the slides, with the speaker notes under each slide

### DISCUSSION THREAD

**Discussion by beyond2060** [quarto-dev/quarto-cli#5849] [Q&A] [answered]
### Description

I want to put revealjs presentations I have created in Quarto on my Quarto website, so that each slide is shown on a single page, with the speaker notes for that slide under each slide. This seems like it's probably something that others have done or wanted - how do I go about doing that?
Thanks in advance for any direction you can give me for this.

**Comment by mcanouil** [comment]
You can't split rendered presentation to get "one slide = one HTML page".
What you can do, is to embed presentation in your website/book or do the other way around, see related discussions:
- #1033
- #1840
- #1433

If you really want to split, you can use tools such as `webshot2` in R to take screenshots of HTML page, in this case your Reveal.js presentation, then include the screenshots.

**Reply by beyond2060** [reply]
Sorry no I didn't mean "one slide = one HTML page", I meant a single page with the slides in sequence down the page, with the presenter notes under each slide.

**Reply by cderv** [reply, marked as answer]
I would say your best chance with Quarto website, is to build a custom Lising page that will layout as you want your different HTML slides. 
https://quarto.org/docs/websites/website-listings-custom.html

You'll have to write HTML content though and come up with how you can manage to present as you want it. 

>  This seems like it's probably something that others have done or wanted

I don't think we encounter yet the question or have seen examples about this.

**Reply by beyond2060** [reply]
Ah yes this looks like it might be a good approach, thanks!

**Comment by beyond2060** [comment]
I've worked out a work-around that isn't too onerous:

- Write the presentation as PPTX
- open in powerpoint 
- export the slides as images
- drag all the images onto an open .qmd file in VS Code when in visual mode. 

This will create a single page containing images of all the slides. You have to add the notes in manually though.

**Reply by cderv** [reply]
Intesting idea ! Glad it works for you !

**Reply by mcanouil** [reply]
On this idea (original code is on a private repository):

The R code render Quarto slides, takes screenshots of each slides and put them in a directory, while the README.qmd takes the images and output them as an HTML table:
<img width="849" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/986f0f39-60d8-4497-be89-05b4f461641d">


A short version to take a screenshot of the first four slides of `slides.qmd`.

```r
input <- "slides.qmd"
n_slides <- 4
quarto::quarto_render(input = input, quiet = TRUE)
for (i in seq_len(n_slides)) {
  webshot2::webshot(
    url = sprintf(
      "file:////%s#%s",
      normalizePath(sub("\\.qmd$", ".html", input)),
      i
    ),
    file = sprintf("contents-%02d.png", i),
    vwidth = 1920,
    vheight = 1005
  )
}
```
